### ISSUE TITLE
ZynAddSubFX Portament not working

### ISSUE THREAD

**Issue by superjosh364** [LMMS/lmms#3936]
The portamento on the ZASFX front page doesn't work for the range 0-64, but if you turn it above 64 it works fine.

**Comment by zonkmachine** [comment]
> but if you turn it above 64 it works fine.

It is portamento there from 64 and up but it's the same level. It sounds like we're sending an on/off message instead of a level value.

**Comment by zonkmachine** [comment]
Right. What we have implemented in lmms is midi control message 65, portamento on/off. There are three other control messages for portamento. 5, 37 and 84. In the midi standard boolean values are realized with a value of 0-127, the lower half of a byte, where 0-63 is off and 64-127 is on.
Midi standard control messages:
https://www.midi.org/specifications/item/table-3-control-change-messages-data-bytes-2
ZASFX, Enable or disable portamento (scroll down for the rest of the portamento code):
https://github.com/LMMS/lmms/blob/stable-1.2/plugins/zynaddsubfx/zynaddsubfx/src/Params/Controller.h#L55:L58

**Comment by musikBear** [comment]
It has been so ever. There has never been a gradual porta control, so this ticket is a feature request, not bug

**Comment by zonkmachine** [comment]
You're probably right there.

**Comment by noahb01** [comment]
Maybe we could replace the knob with a toggle? I assume there is some technical reason for implementing the portamento control as an on/off instead of a gradual control.

**Comment by superjosh364** [comment]
Well, if you have a portament knob, it usually determines the amount of time the pitch glides from one note to another. So there should be a toggle switch to turn it on, and then a knob for adjusting the length of the portamento.

**Comment by Spekular** [comment]
From what I've seen, there tends to be a single knob w/ no toggle. Setting
the portamento time to 0 naturally turns it off.

**Comment by noahb01** [comment]
@Spekular @superjosh364 What I mean is that if there is no way to access the actual portamento value, we should replace it with an on/off toggle. If we can control that value, by all means do it.

**Comment by musikBear** [comment]
its just because the 'wrong' controller (red circle) were selected for the ui
![zynporta](https://user-images.githubusercontent.com/6662066/36063809-060fbf42-0e82-11e8-9f27-3878b9bea75a.JPG)
It should be *time.t* (green circle).
But as you can see, there are more advanced porta-settings available.
Perhaps the dial could be preserved, but default 'grayed-out' meaning *no-porta*. if enabled with left-click, *time.t* could be changed.
That way the current ui would not have to change drastically.

There is also all the realstate mentioned in #1965

**Comment by zonkmachine** [comment]
Poking this one a bit in: https://github.com/LMMS/lmms/pull/4167
